pulsar/producer_impl.go

```diff
@@ -87,6 +88,9 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 	if options.BatchingMaxPublishDelay <= 0 {
 		options.BatchingMaxPublishDelay = defaultBatchingMaxPublishDelay
 	}
+	if options.PartitionsAutoDiscoveryInterval == 0 {
```

Comment:
What happens if a negative value is passed in?

Reply:
Thanks for the catch. Updated the check to include negative values.